dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -537,6 +529,21 @@ public GHRepository getRepository(String repositoryId) {
         }
     }
 
+    private String getCommitSHA(GHRef ref, GHRepository repository, String refName) throws IOException {
+        String sha = ref.getObject().getSha();
+        if (ref.getObject().getType().equals("tag")) {
+            GHTagObject tagObject = repository.getTagObject(sha);
+            sha = tagObject.getObject().getSha();
+        } else if (ref.getObject().getType().equals("branch")) {
+            GHBranch branch = repository.getBranch(refName);
+            sha = branch.getSHA1();
+        } else {
+            // I'm not sure when this would ever happen.
+            throw new CustomWebApplicationException("Unsupported branch/tag/release. Unable to find commit ID.", HttpStatus.SC_BAD_REQUEST);
```

Comment:
Slightly wary of this for a hotfix; if this `else` ever does execute, the code on line 516 will now throw an exception whereas before it just used the sha from the old line 517. I think you should just log it here and not throw an exception.

Reply:
Changed.